pkg/providers/vsphere/vmlifecycle/update_status_test.go

```diff
@@ -3242,9 +3242,11 @@ var _ = Describe("Snapshot status", func() {
 				It("should update the current snapshot status to match the found snapshot", func() {
 					Expect(vmlifecycle.ReconcileStatus(vmCtx, ctx.Client, vcVM, data)).To(Succeed())
 					Expect(vmCtx.VM.Status.CurrentSnapshot).ToNot(BeNil())
-					Expect(vmCtx.VM.Status.CurrentSnapshot.APIVersion).To(Equal(vmSnapshot.APIVersion))
-					Expect(vmCtx.VM.Status.CurrentSnapshot.Kind).To(Equal(vmSnapshot.Kind))
-					Expect(vmCtx.VM.Status.CurrentSnapshot.Name).To(Equal(vmSnapshot.Name))
+					Expect(vmCtx.VM.Status.CurrentSnapshot.Type).To(Equal(vmopv1.VirtualMachineSnapshotReferenceTypeManaged))
+					Expect(vmCtx.VM.Status.CurrentSnapshot.SnapshotReference).ToNot(BeNil())
+					Expect(vmCtx.VM.Status.CurrentSnapshot.SnapshotReference.APIVersion).To(Equal(vmSnapshot.APIVersion))
+					Expect(vmCtx.VM.Status.CurrentSnapshot.SnapshotReference.Kind).To(Equal(vmSnapshot.Kind))
+					Expect(vmCtx.VM.Status.CurrentSnapshot.SnapshotReference.Name).To(Equal(vmSnapshot.Name))
 				})
 			})
```

Comment:
The test below here `When("the snapshot custom resource doesn't exist", func() {`, it should expect a non empty `CurrentSnapshot ` after this change I assume?


I think the test is incorrect because it doesn't actually removes the vmSnapshot since it has finalizer
```
// Set a finalizer
vmSnapshot.Finalizers = append(vmSnapshot.Finalizers, "random-finalizer")
Expect(ctx.Client.Update(ctx, vmSnapshot)).To(Succeed())
// Mark the snapshot for deletion. The finalizer will
// make sure the object is not garbage collected.
Expect(ctx.Client.Delete(ctx, vmSnapshot)).To(Succeed())`
```

Reply:
Corrected the test logic. In "the snapshot custom resource doesn't exist", the object was not deleted because of the vmsnapshot finalizer in the dummy object.